homeassistant/components/smartthings/media_player.py

```diff
@@ -0,0 +1,196 @@
+"""Support for media players through the SmartThings cloud API."""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import Any
+
+from pysmartthings import Capability, DeviceEntity
+
+from homeassistant.components.media_player import (
+    DOMAIN as MEDIA_PLAYER_DOMAIN,
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_BROKERS, DOMAIN
+from .entity import SmartThingsEntity
+
+VALUE_TO_STATE = {
+    "buffering": MediaPlayerState.BUFFERING,
+    "pause": MediaPlayerState.PAUSED,
+    "paused": MediaPlayerState.PAUSED,
+    "play": MediaPlayerState.PLAYING,
+    "playing": MediaPlayerState.PLAYING,
+    "stop": MediaPlayerState.IDLE,
+    "stopped": MediaPlayerState.IDLE,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add media players for a config entry."""
+    broker = hass.data[DOMAIN][DATA_BROKERS][config_entry.entry_id]
+    async_add_entities(
+        [
+            SmartThingsMediaPlayer(device)
+            for device in broker.devices.values()
+            if broker.any_assigned(device.device_id, MEDIA_PLAYER_DOMAIN)
+        ]
+    )
+
+
+def get_capabilities(capabilities: Sequence[str]) -> Sequence[str] | None:
+    """Return all capabilities supported if minimum required are present."""
+    supported = [
+        Capability.audio_mute,
+        Capability.audio_volume,
+        Capability.media_input_source,
+        Capability.media_playback,
+        Capability.switch,
+    ]
+    # Must have one of these.
+    media_player_capabilities = [
+        Capability.audio_mute,
+        Capability.audio_volume,
+        Capability.media_input_source,
+        Capability.media_playback,
+    ]
+    if any(capability in capabilities for capability in media_player_capabilities):
+        return supported
+    return None
+
+
+class SmartThingsMediaPlayer(SmartThingsEntity, MediaPlayerEntity):
+    """Define a SmartThings media player."""
+
+    def __init__(self, device: DeviceEntity) -> None:
+        """Initialize the media_player class."""
+        super().__init__(device)
+        self._state = None
+        self._state_attrs = None
```

Comment:
Unused. Not sure if they are required by the parent entity

Reply:
Good catch, they are not required at all!